drag_and_drop_v2/drag_and_drop_v2.py

```diff
@@ -412,12 +382,120 @@ def default_background_image_url(self):
         """ The URL to the default background image, shown when no custom background is used """
         return self.runtime.local_resource_url(self, "public/img/triangle.png")
 
+    @property
+    def attemps_remain(self):
+        return self.max_attempts is None or self.num_attempts < self.max_attempts
+
     @XBlock.handler
     def get_user_state(self, request, suffix=''):
         """ GET all user-specific data, and any applicable feedback """
         data = self._get_user_state()
         return webob.Response(body=json.dumps(data), content_type='application/json')
 
+    def _drop_item_standard(self, item_attempt):
+        item = self._get_item_definition(item_attempt['val'])
+
+        is_correct = self._is_attempt_correct(item_attempt)  # Student placed item in a correct zone
+        if is_correct:  # In standard mode state is only updated when attempt is correct
+            self.item_state[str(item['id'])] = self._make_state_from_attempt(item_attempt, is_correct)
+
+        self._mark_complete_and_publish_grade()
+        self._publish_item_dropped_event(item_attempt, is_correct)
+
+        item_feedback_key = 'correct' if is_correct else 'incorrect'
+        item_feedback = item['feedback'][item_feedback_key]
+        overall_feedback = self.data['feedback']['finish'] if self._is_finished() else None
+
+        return {
+            'correct': is_correct,
+            'finished': self._is_finished(),
+            'overall_feedback': overall_feedback,
+            'feedback': item_feedback
+        }
+
+    def _drop_item_assessment(self, item_attempt):
+        if not self.attemps_remain:
+            raise JsonHandlerError(409, _("Max number of attempts reached"))
+
+        item = self._get_item_definition(item_attempt['val'])
+
+        is_correct = self._is_attempt_correct(item_attempt)
+        # State is always updated in assessment mode, to make do_attempt work correctly
+        self.item_state[str(item['id'])] = self._make_state_from_attempt(item_attempt, is_correct)
+
+        self._publish_item_dropped_event(item_attempt, self._is_attempt_correct(item_attempt))
```

Comment:
`self._is_attempt_correct(item_attempt)` is already invoked a couple of lines above, you can replace the second parameter with `is_correct` variable.